kubernetes/token.go

```diff
@@ -1,16 +1,23 @@
 package kubernetes
 
-import "io/ioutil"
+import (
+	"io/ioutil"
+	"time"
+
+	kialiConfig "github.com/kiali/kiali/config"
+)
 
 // Be careful with how you use this token. This is the Kiali Service Account token, not the user token.
 // We need the Service Account token to access third-party in-cluster services (e.g. Grafana).
 
-const DefaultServiceAccountPath = "/var/run/secrets/kubernetes.io/serviceaccount/token"
+var DefaultServiceAccountPath = "/var/run/secrets/kubernetes.io/serviceaccount/token"
 
 var KialiToken string
+var tokenRead time.Time
+var tokenExpireDuration time.Duration
 
 func GetKialiToken() (string, error) {
-	if KialiToken == "" {
+	if KialiToken == "" || IsTokenExpired() {
```

Comment:
It'd be nice to refresh before the token expires by watching for when the token changes and updating `KialiToken` then. We'd need a separate watch thread that continuously reads the token from the service account path and diff the existing token with the token it reads. The cache (or other places we use KialiToken) can then watch for when KialiToken changes. That may require some refactoring though since KialiToken is a global var and it doesn't seem like a good idea to startup a background goroutine simply by importing this package. That can be an optimization we address later. Could you add a `TODO:` here to refresh the token when it changes rather than after it expires?

Reply:
Added TODO comment

Reply:
Yep that'd be good